### ISSUE TITLE
Bug Report：Incorrect optimizer state loading

### ISSUE THREAD

**Issue by ironhxs** [HaoWang420/Gradient-guided-Modality-Decoupling#6]
## Description
I encountered a bug in the code where the optimizer state is not being loaded correctly from the checkpoint. 
## File Location
**File:** `trainer/trainer.py`  
**Line:** 124
## Current Code (Incorrect)
```python
self.optimizer.load_state_dict = checkpoint['optimizer']   # Line 124
```
## Problem
There's a syntax error - missing parentheses when calling ```load_state_dict```. The current code assigns the checkpoint dictionary to the method itself rather than calling the method with the checkpoint data as an argument.
## Solution
```python
self.optimizer.load_state_dict(checkpoint['optimizer'])   # Fixed
```
